lib/src/merged_tree.rs

```diff
@@ -1194,7 +1194,17 @@ impl MergedTreeBuilder {
             MergedTreeId::Merge(base_tree_ids) => base_tree_ids,
         };
         let new_tree_ids = self.write_merged_trees(base_tree_ids, store)?;
-        Ok(MergedTreeId::Merge(new_tree_ids.simplify()))
+        match new_tree_ids.simplify().into_resolved() {
+            Ok(single_tree_id) => Ok(MergedTreeId::resolved(single_tree_id)),
+            Err(tree_id) => {
+                let MergedTree::Merge(tree) = store.get_root_tree(&MergedTreeId::Merge(tree_id))?
+                else {
+                    unreachable!();
+                };
+                let merged = merge_trees(&tree)?;
+                Ok(MergedTree::Merge(merged.simplify()).id())
```

Comment:
nit: does it make sense to copy debug assertion from (or extract bottom half of) `MergedTree::merge()`?

Reply:
Good point. Done by moving the code into `MergedTree::resolve()`.